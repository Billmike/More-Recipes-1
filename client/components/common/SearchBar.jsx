import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  id: PropTypes.string,
  search: PropTypes.string.isRequired,
  list: PropTypes.string,
  handleSearchChange: PropTypes.func.isRequired,
  handleSubmitSearch: PropTypes.func.isRequired,
};

const defaultProps = {
  id: null,
  list: undefined,
};

/**
 * Function representing search bar component
 *
 * @param {object} props - The properties passed to the component
 *
 * @returns {any} React element
 */
const SearchBar = (props) => {
  const { search, list, handleSearchChange, handleSubmitSearch } = props;

  return (
    <div id={props.id} className="row">
      <div className="col s12 m3 l3 push-m9 push-l9">
        <select
          name="searchBy"
          id="search-by"
          className="browser-default search-by z-depth-1"
          value={search}
          onChange={handleSearchChange}
        >
          <option value="title">Search By Title</option>
          <option value="ingredients">Search By Ingredients</option>
        </select>
      </div>
      <div
        className="col s12 m6 l6 offset-m3 offset-l3 pull-m3 pull-l3 search-bar"
      >
        <form onSubmit={handleSubmitSearch}>
          <div className="row">
            <div className="input-field white">
              <input
                id="search-input"
                type="search"
                placeholder="Search for recipes"
                value={list}
                onChange={handleSearchChange}
                required
              />
              <a
                role="button"
                tabIndex="0"
                className="label-icon"
                htmlFor="search"
                onClick={handleSubmitSearch}
              >
                <i className="material-icons">search</i>
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  );
};

SearchBar.propTypes = propTypes;
SearchBar.defaultProps = defaultProps;

export default SearchBar;
