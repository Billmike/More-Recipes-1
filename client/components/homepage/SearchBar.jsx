import React from 'react';

const SearchBar = () => (
  <div className="row">
    <div id="search-bar" className="col s12 m6 l6 offset-m3 offset-l3">
      <form className="container">
        <div className="input-field white">
          <input
            id="search-input"
            type="search"
            placeholder="Search for recipes"
            required
          />
          <label className="label-icon" htmlFor="search">
            <i className="material-icons">search</i>
          </label>
        </div>
      </form>
    </div>
  </div>
);

export default SearchBar;
