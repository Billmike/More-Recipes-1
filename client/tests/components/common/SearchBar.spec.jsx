import React from 'react';

import SearchBar from '../../../components/common/SearchBar';

describe('Search bar component', () => {
  const props = {
    search: 'ingredients',
    list: 'palm oil, beans',
    handleSearchChange: jest.fn(),
    handleSubmitSearch: jest.fn(),
  };
  it('should render component', () => {
    const wrapper = shallow(<SearchBar {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate change event', () => {
    const wrapper = shallow(<SearchBar {...props} />);
    wrapper.find('#search-input').simulate('change');
    expect(props.handleSearchChange).toBeCalled();
  });
  it('should simulate click event', () => {
    const wrapper = shallow(<SearchBar {...props} />);
    wrapper.find('#search-button').simulate('click');
    expect(props.handleSubmitSearch).toBeCalled();
  });
});
