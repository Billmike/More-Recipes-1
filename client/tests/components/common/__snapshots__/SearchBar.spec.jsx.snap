// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search bar component should render component 1`] = `
<div
  className="row"
  id={null}
>
  <div
    className="col s12 m3 l3 push-m9 push-l9"
  >
    <select
      className="browser-default search-by z-depth-1"
      id="search-by"
      name="searchBy"
      onChange={[MockFunction]}
      value="ingredients"
    >
      <option
        value="title"
      >
        Search By Title
      </option>
      <option
        value="ingredients"
      >
        Search By Ingredients
      </option>
    </select>
  </div>
  <div
    className="col s12 m6 l6 offset-m3 offset-l3 pull-m3 pull-l3 search-bar"
  >
    <form
      onSubmit={[MockFunction]}
    >
      <div
        className="row"
      >
        <div
          className="input-field white"
        >
          <input
            id="search-input"
            onChange={[MockFunction]}
            placeholder="Search for recipes"
            required={true}
            type="search"
            value="palm oil, beans"
          />
          <a
            className="label-icon"
            htmlFor="search"
            id="search-button"
            onClick={[MockFunction]}
            role="button"
            tabIndex="0"
          >
            <i
              className="material-icons"
            >
              search
            </i>
          </a>
        </div>
      </div>
    </form>
  </div>
</div>
`;
